functions/Invoke-DbaLogShippingRecovery.ps1

```diff
@@ -0,0 +1,390 @@
+        [Parameter(Mandatory=$false, Position=7)][int]$Delay = 5
+	)
+
+    BEGIN
+    {
+        Write-Message -Message "Attempting to connect to Sql Server.." -Level 2 -Silent $Silent
+        try 
+        {
+            $server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $SqlCredential
+        }
+        catch 
+        {
+            Stop-Function -Message "Could not connect to Sql Server instance" -Silent $Silent -InnerErrorRecord $_ -Target $SqlServer 
+            return
+        }
+		
+    }
+
+    PROCESS
+    {
+        Write-Message -Message "Started Log Shipping Recovery" -Level 2 -Silent $Silent
+
+        #region Query setup
+        # Query for retrieving the log shipping information
+        $query = "
+----------------------------------------------------------------
+SELECT  lss.primary_server, lss.primary_database, lsd.secondary_database, lss.backup_source_directory,
+        lss.backup_destination_directory, lss.last_copied_file, lss.last_copied_date,
+        lsd.last_restored_file, sj1.name AS 'copyjob', sj2.name AS 'restorejob'
+FROM    msdb.dbo.log_shipping_secondary AS lss
+        INNER JOIN msdb.dbo.log_shipping_secondary_databases AS lsd ON lsd.secondary_id = lss.secondary_id
+		INNER JOIN msdb.dbo.sysjobs AS sj1 ON sj1.job_id = lss.copy_job_id
+		INNER JOIN msdb.dbo.sysjobs AS sj2 ON sj2.job_id = lss.restore_job_id "
+        
+        # Check if they commandlet is eecuted from a pipeline
+		if ($PSCmdlet.MyInvocation.ExpectingInput) 
+		{
+			$db = $_
+            
+            $query += "WHERE lsd.secondary_database = '$db' "
+		}
+		else
+		{
+            # If just one database is set in the database parameter
+			if($Database -ne $null)
+            {
+			    $db = $Database[0]
+
+                $query += "WHERE lsd.secondary_database = '$db'"
+            }
+		}
+        
+        # Add an extra line to easier read the query when in verbose mode
+        $query +="
+----------------------------------------------------------------"
+        #endregion Query setup
+
+        # Retrieve the log shipping information from the secondary instance
+        try
+        {
+            Write-Message -Message "Retrieving log shipping information from the secondary instance" -Level 5 -Silent $Silent
+            $logshipping_details = Invoke-Sqlcmd2 -ServerInstance $SqlServer -Database 'msdb' -Query $query 
+        }
+        catch
+        {
+            Stop-Function -Message ("Error retrieving the log shipping details: " + $_.Exception.Message) -Silent $Silent -InnerErrorRecord $_ -Target $SqlServer
+            
+            return
+        }
+        
+        # Checking the status of the SQL Server Agent service
+        Write-Message -Message ("Checking the status of the SQL Server Agent" + $ls.secondary_database + "'") -Level 2 -Silent $Silent
+        $agentservice = Get-Service | Where-Object -like -value '*sql*agent*' -Property 'name'
```

Comment:
Hey Sander,
This line will only account for 

- services on localhost
- any service named sqlagent

It does not handle: remote servers or multiple instances.